nexus/src/app/oximeter.rs

```diff
@@ -28,6 +28,12 @@ use std::num::NonZeroU32;
 use std::time::Duration;
 use uuid::Uuid;
 
+/// How long a metrics producer remains registered to a collector.
+///
+/// Producers are expected to renew their registration lease periodically, at
+/// some interval of this overall duration.
+pub const PRODUCER_LEASE_DURATION: Duration = Duration::from_secs(10 * 60);
```

Comment:
I'm working on the producer-renewal side of things, and I think we should find a way to make this public and easily-accessible in `oximeter-producer`. The producer needs to renew within some fraction of this time, and I'm a bit worried that we'll change this and forget to change that renewal period. `oximeter-producer` already depends on the grab-bag that is `omicron-common`, so that might be a good place.

Reply:
I think at one point we mentioned having nexus return this when a producer registers - if we did that, would it remove the need to share the constant? I share your worry, but am also concerned about a future world where we can update services independently; a change to a constant like this wouldn't be an API break we'd detect at the progenitor level, so it could sneak in. But if Nexus told producers what to use, we'd only have to worry about changing it in Nexus, I think?